Sources/SourceKitLSP/Swift/OpenInterface.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SourceKitD
+import LanguageServerProtocol
+import LSPLogging
+
+struct InterfaceInfo {
+  var contents: String
+}
+
+private var textualInterfaces: [String:DocumentURI] = [:]
+
+extension SwiftLanguageServer {
+  public func openInterface(_ request: LanguageServerProtocol.Request<LanguageServerProtocol.OpenInterfaceRequest>) {
+    let uri = request.params.textDocument.uri
+    let moduleName = request.params.name
+    self.queue.async {
+      
+      if let uri = textualInterfaces[moduleName] {
+        request.reply(.success(InterfaceDetails(uri: uri)))
+        return
+      }
+      
+      let interfaceFilePath = self.generatedInterfacesPath.appendingPathComponent("\(moduleName).swift")
+      let interfaceDocURI = DocumentURI(interfaceFilePath)
+      self._openInterface(request, uri, moduleName, interfaceDocURI) { result in
+        switch result {
+        case .success(let interfaceInfo):
+          do {
+            try interfaceInfo.contents.write(to: interfaceFilePath, atomically: true, encoding: String.Encoding.utf8)
+            textualInterfaces[moduleName] = interfaceDocURI
+            request.reply(.success(InterfaceDetails(uri: interfaceDocURI)))
+          } catch {
+            request.reply(.failure(ResponseError.unknown(error.localizedDescription)))
+          }
+        case .failure(let error):
+          log("open interface failed: \(error)", level: .warning)
+          request.reply(.failure(ResponseError(error)))
+        }
+      }
+    }
+  }
+  
+  private func _openInterface(_ request: LanguageServerProtocol.Request<LanguageServerProtocol.OpenInterfaceRequest>,
+                              _ uri: DocumentURI,
+                              _ name: String,
+                              _ interfaceURI: DocumentURI,
+                              _ completion: @escaping (Swift.Result<InterfaceInfo, SKDError>) -> Void) {
```

Comment:
Could you add descriptions for these parameters? Also: Is there any particular reason why you chose to not use argument labels for any of these? I personally always prefer argument label.